### ISSUE TITLE
Public API needs a naming pass

### ISSUE THREAD

**Issue by caprica** [caprica/vlcj#716]
Review especially method names.

**Comment by caprica** [comment]
Also, because of the way things are implemented now we can end up with some unfortunate method chaining, like:
```
List<String> subItems = mediaPlayer.media().get().subitems().get().items().mrls();
```
Some of this is inevitable, but it's worth a review.

**Comment by caprica** [comment]
- [ ] uk.co.caprica.vlcj
 - [ ] callbackmedia
  - [ ] nonseekable
  - [ ] seekable
 - [ ] component
  - [ ] callback
  - [ ] overlay
 - [ ] condition
  - [ ] media
  - [ ] mediaplayer
 - [ ] discoverer
 - [ ] discovery
  - [ ] provider
  - [ ] strategy
 - [ ] enums
 - [ ] eventmanager
 - [ ] factory
 - [ ] log
 - [ ] media
 - [ ] medialist
 - [ ] model
 - [ ] player
  - [ ] base
  - [ ] embedded
  - [ ] list
 - [ ] renderer
 - [ ] version

**Comment by caprica** [comment]
Try and reduce the numbers of these services/aspects. Some of them have only a single method.

**Comment by caprica** [comment]
Maybe remove interfaces for things like MediaPlayer.

I think interfaces *may* come back as vlcj-pro may require it, but for now they add nothing, and this has to be refactored anyway because of the services/aspects classes we have now.

**Comment by caprica** [comment]
If we're going to keep user-data on the high level components, it should not be a service and probably should just be on the component class itself.

**Comment by caprica** [comment]
User data services have been removed and re-implemented on the component themselves.

**Comment by caprica** [comment]
Remaining review tasks:

[ ] "service" vs "aspect" vs something else?
[ ] interfaces

**Comment by caprica** [comment]
On interfaces, there is no point in forcing the use of an interface for the sake of it - in particular vlcj core has no need for common interfaces.

The issue is that vlcj-pro needs to mimic the vlcj core API so implementing a common interface would in theory be useful.

However, even in vlcj-pro, there is no absolute need for a *shared* interface, the vlcj-pro media player can mimic the vlcj API without having to implement the same interfaces. With interfaces, it would have meant interfaces not just for high level components like MediaPlayer, but also for all of the "services" it exposes like MediaService, VideoSurfaceService, ControlsService, MarqueeService, LogoService and so on.

Using interfaces would of course guarantee the same method signatures, which would undoubtedly be convenient, but not at the expense of cluttering the vlcj core project with interfaces that it itself has no use for.

So, in short, the various MediaPlayer interfaces will be removed from vlcj itself, vlcj-pro will re-implement the vlcj API simply by mirroring the API and class structures, not be implementing shared interfaces.

**Comment by caprica** [comment]
Media player interfaces removed at ef9ccf0d63c02747b8f8e365ece697ab75380251.

**Comment by caprica** [comment]
Examples:

 - ControlsService, MediaService, MetaDataService
 - ControlsAspect, MediaAspect, MetaDataAspect
 - ControlsApi, MediaApi, MetaDataApi

Despite the awkward capitalisation, I think I prefer the Api suffix.

It only affects class-names, so it's no big deal really.

**Comment by caprica** [comment]
Renamed Services to Api at 12e519ab90c12eeeb3f8d8ee234afe63a2bd35d9.
